std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -235,6 +247,45 @@ public static long char_length(String str) {
     return str.length();
   }
 
+  /**
+   * Gets the length of the string in graphemes
+   *
+   * @param str the string to measure
+   * @return length of the string
+   */
+  public static long grapheme_length(String str) {
+    BreakIterator iter = BreakIterator.getCharacterInstance();
+    iter.setText(str);
+    long len = 0;
+    while (iter.next() != BreakIterator.DONE) {
+      len++;
+    }
+    return len;
+  }
+
+  /** Returns a prefix of the string not exceeding the provided grapheme length. */
+  public static String take_prefix(String str, long grapheme_length) {
```

Comment:
`take_prefix` and `take_suffix` could potentially use one implementation (with appropriate parameters) for DRY

Reply:
I don't see that. The logic is similar but different in almost every place (with prefix you do next(len), with suffix you do last+next(-len), with prefix you substring 0-current and with suffix current-length - the only part that is really the same is initialization of the break iterator.
I can't imagine how I can rewrite this to a single function and make the code simpler and not more complicated - although I'm not saying its impossible so I'm open to a suggestion how it may look like. But personally I don't see how I can improve this.